ComfortZoneApp/app/src/main/java/com/example/comfortzone/flight/utils/FlightUtil.java

```diff
@@ -0,0 +1,31 @@
+package com.example.comfortzone.flight.utils;
+
+import static com.example.comfortzone.flight.ui.FlightFragment.LOC_IATA;
+
+import com.example.comfortzone.flight.callbacks.FlightBookingsCallback;
+import com.example.comfortzone.flight.data.BookingClient;
+import com.example.comfortzone.flight.models.Bookings;
+import com.example.comfortzone.models.WeatherData;
+
+import rx.Observable;
+import rx.Subscriber;
+
+public class FlightUtil {
+
+    public static Observable getDeepLink(WeatherData cityData) {
+        return Observable.create(new Observable.OnSubscribe<Object>() {
+            @Override
+            public void call(Subscriber<? super Object> subscriber) {
+                BookingClient client = new BookingClient();
+                client.getBookingLinks(LOC_IATA, cityData.getIata(), new FlightBookingsCallback() {
```

Comment:
yeah I don't think we should use LOC_IATA static variable to get the IATA, you should be passing it as param from whoever calls this method..

On top of mind, you should do:
1. migrate your local variables to HostActivity
2. define an interface for HostActivity to provide those values
3. from the fragment, you can get those values by getting the parent activity and casting it to the interface
4. pass the values around where you need it, in this case pass it as intent argument to `CityDetailActivity`